mobile_packet_verifier/src/banning/purger.rs

```diff
@@ -0,0 +1,50 @@
+use std::time::Duration;
+
+use chrono::{DateTime, Utc};
+use futures::FutureExt;
+use sqlx::PgPool;
+use task_manager::ManagedTask;
+
+pub struct BanPurger {
+    pool: PgPool,
+    interval: Duration,
+}
+
+impl ManagedTask for BanPurger {
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> futures::future::LocalBoxFuture<'static, anyhow::Result<()>> {
+        self.run(shutdown).boxed_local()
+    }
+}
+
+impl BanPurger {
+    pub fn new(pool: PgPool, interval: Duration) -> Self {
+        Self { pool, interval }
+    }
+
+    async fn run(mut self, mut shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("ban purger starting");
+        let mut timer = tokio::time::interval(self.interval);
+
+        loop {
+            tokio::select! {
+                biased;
+                _ = &mut shutdown => break,
+                _ = timer.tick() => self.purge(Utc::now()).await?
```

Comment:
purging bans based on `Utc::now()` seems wrong, we could be behind in processing incoming data transfer sessions and be eliminating bans that are still valid based on where we are with processing the incoming data transfer queue

Reply:
I missed that you offset Utc::now() by 7 days, nevermind this comment